Source/CombatExtended/CombatExtended/Projectiles/ProjectileCE.cs

```diff
@@ -324,26 +324,28 @@ private float GravityFactor
             }
         }
 
+
+        // Todo: When rimworld harmony updates to 2.0.4 use FieldRefAccess
+        private static readonly FieldInfo subGraphics = typeof(Graphic_Collection).GetField("subGraphics", BindingFlags.NonPublic | BindingFlags.Public | BindingFlags.Instance);
+
         private Material[] shadowMaterial;
         private Material ShadowMaterial
         {
             get
             {
-                if (shadowMaterial == null)
+                if (shadowMaterial != null) return shadowMaterial[Rand.Range(0, this.shadowMaterial.Length)];
```

Comment:
Inverting this check requires duplicating the return line.  While early return is nice, "if cached is null, set cache, return cache" is a very common pattern for a reason.